hibernate-core/src/main/java/org/hibernate/collection/internal/AbstractPersistentCollection.java

```diff
@@ -620,6 +624,9 @@ public final boolean unsetSession(SharedSessionContractImplementor currentSessio
 		prepareForPossibleLoadingOutsideTransaction();
 		if ( currentSession == this.session ) {
 			if ( !isTempSession ) {
+				if ( hasQueuedOperations() ) {
+					LOG.queuedOperationWhenDetachFromSession( MessageHelper.collectionInfoString( getRole(), getKey() ) );
+				}
```

Comment:
Logs a warning if a collection is detached with queued operations. 

This warning is known to be logged in the following situations, as illustrated by running the unit tests:
* the application explicitly detaches the collection owner entity while it has queued operations;
* Hibernate detaches the collection with queued operations after rolling back the transaction.